Shared/Samples/Search for web map/SearchForWebMapView.Model.swift

```diff
@@ -0,0 +1,121 @@
+            task = Task {
+                let parameters = try queryParameters(for: query)
+                let results = try await findItems(using: parameters)
+                
+                await MainActor.run {
+                    portalItems = results
+                    isLoadingResults = false
+                }
+            }
+        }
+
+        /// Finds the portal items that match the next query parameters from the previous search.
+        func findNextItems() throws {
+            guard let nextQueryParameters else { return }
+            // Cancel the current search operation if there is one.
+            task?.cancel()
+            isLoadingResults = false
+
+            // Find the next results.
+            isLoadingResults = true
+            task = Task {
+                let nextResults = try await findItems(using: nextQueryParameters)
+                
+                await MainActor.run {
+                    portalItems.append(contentsOf: nextResults)
+                    isLoadingResults = false
+                }
+            }
+        }
+        
+        /// Finds the portal items that match the given query parameters.
+        /// - Parameter queryParameters: The portal query parameters to find the portal items.
+        /// - Returns: The portal items that were found.
+        private func findItems(using queryParameters: PortalQueryParameters) async throws -> [PortalItem] {
+            // Get the results from the portal using the parameters.
+            let resultsSet = try await portal.findItems(queryParameters: queryParameters)
+            nextQueryParameters = resultsSet.nextQueryParameters
+            return resultsSet.results
+        }
+        
+        /// The portal query parameters for a given query.
+        /// - Parameter query: The text query used to create the parameters.
+        /// - Returns: A new `PortalQueryParameters` object.
+        private func queryParameters(for query: String) throws -> PortalQueryParameters {
+            // Create a date string for a date range to search within.
+            // Note: Web maps authored prior to July 2nd, 2014 are not supported.
+            let startDate = try Date(
+                "July 2, 2014",
+                strategy: Date.FormatStyle().day().month().year().parseStrategy
+            )
+            let dateRange = startDate.millisecondsSince1970...Date.now.millisecondsSince1970
+            let dateString = "uploaded:[\(dateRange.lowerBound) TO \(dateRange.upperBound)]"
+            
+            // Create a string to filter for web maps.
+            let typeString = "type:\"Web Map\""
+            
+            // Create the portal query parameters with the strings.
+            return PortalQueryParameters(query: "\(query) AND \(typeString) AND \(dateString)")
+        }
+    }
+}
+
+private extension Date {
+    /// The milliseconds between the date value and 00:00:00 UTC on 1 January 1970.
+    var millisecondsSince1970: Int64 {
```

Comment:
I was kinda surprised to find out this is the standard way to do date search with ArcGIS REST API. Probably worth paraphrasing from the source https://developers.arcgis.com/rest/users-groups-and-items/search-reference.htm#ESRI_SECTION1_B20F9637ADF04AD2B88172D43AA6FC42 to add some comment on why this is needed.